tests/unittests/client/test_experiment_client.py

```diff
@@ -676,6 +672,9 @@ def amnesia(num=1):
 
             assert len(experiment.fetch_trials()) == 1
 
+            with pytest.raises(WaitingForTrials):
```

Comment:
It's a bit confusing now. It could just return [new_value] (without tacking track of a count). First suggest works, next ones will lead to WaitingForTrials.

Reply:
Actually, to test ReservationRaceCondition we could instead mock producer.produce to just to nothing and return 1.

Reply:
Ah, it's done at test_suggest_reserve_race_condition. Ok, please forget my second comment. :)

Reply:
the first suggestion is not working, if the amnesia.count  is not tracked WaitingForTrials is not raised

Reply:
Right, it should only return [0] for WaitingForTrials to be raised. Since that's what we are testing (the fact that it is raising that error) we could only do this and drop the 4 lines below testing the suggest with `new_value`.

Reply:
then the second suggest will also raise WaitingForTrials

Reply:
Yeah, no need to test for second suggest.

Reply:
done